x-pack/plugins/security_solution/public/common/hooks/use_invalid_filter_query.tsx

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useEffect } from 'react';
+import { useAppToasts } from './use_app_toasts';
+
+/**
+ * Adds a toast error message whenever invalid KQL is submitted through the search bar
+ */
+export const useInvalidFilterQuery = ({
+  filterQuery,
+  kqlError,
+}: {
+  filterQuery?: string;
+  kqlError?: Error;
+}) => {
+  const { addError } = useAppToasts();
+
+  useEffect(() => {
+    if (filterQuery === undefined && kqlError != null) {
+      addError(kqlError, { title: kqlError.name });
+    }
+    // This disable is required to only trigger the toast once per render
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [filterQuery, addError]);
```

Comment:
I think you can go for it and add `addError` as a dependency, because you won't have `kqlError != null`